src/Application.php

```diff
@@ -285,15 +284,17 @@ protected function doRunCommand(Command $command, InputInterface $input, OutputI
                 if (OutputInterface::VERBOSITY_VERBOSE <= $output->getVerbosity()) {
                     throw $e;
                 }
+
+                return $event->getExitCode();
             } else {
                 throw $event->getException();
             }
-        } finally {
-            $event = new ConsoleTerminateEvent($command, $input, $output, $exitCode);
-            $this->dispatcher->dispatch(ConsoleEvents::TERMINATE, $event);
         }
 
-        return $exitCode;
+        $event = new ConsoleTerminateEvent($command, $input, $output, $exitCode);
+        $this->dispatcher->dispatch(ConsoleEvents::TERMINATE, $event);
+
+        return $event->getExitCode();
```

Comment:
No I actually reverted that because the exitCode can be changed with ConsoleTerminateEvent, and the variable would become outdated.

This is the same way its done in the parent class, so we should follow that for compatibility :)